### ISSUE TITLE
hx-trigger="click from:#another" does not work

### ISSUE THREAD

**Issue by BoPeng** [bigskysoftware/htmx#392]
I would like to trigger two `hx-post` events to send data to two different URLs with a single click. It boils down to the following code

```html
<!DOCTYPE html>
<html lang="en">
<body>
	<a href="#" hx-get="first.html" id="first">fist link</a>
	<a href="#" hx-get="second.html"
            hx-trigger="click from:#first">second link</a>
</body>
<script src="https://unpkg.com/htmx.org@1.2.1"></script>
</html>
```

However, when I click the first link, the second link is not triggered. Experiments show that

1. `hx-trigger="change from:#first"` does not work
2. `hx-trigger="mouseup from:#first"` works

`mouseup` is close to what I need but still wondering why `click from:#first` does not work.

**Comment by 1cg** [comment]
Hi Bo,

We have to preventDefault() on anchor tags to prevent the browser from doing navigation, and i have a feeling this is leading to what you are seeing.

https://github.com/bigskysoftware/htmx/blob/5a68871ac388654e9e381d4655abfc1d17f8804b/src/htmx.js#L925

We have to do the same things on forms and submit buttons within forms.

**Comment by BoPeng** [comment]
This might be related to #388 since in this particular case, I have `<a href="#"> `  so `elt.href.indecOf('#")` is not zero only because `elt.href` has been expanded.

**Comment by 1cg** [comment]
Can you verify against the dev version?  I have applied your fix.

**Comment by 1cg** [comment]
ah, just saw your next comment.

Can you debug to see if the click is being fired and filtered for some reason?

**Comment by BoPeng** [comment]
Tried but did not figure out how hx-trigger is supposed to work. I will try again.

**Comment by BoPeng** [comment]
OK, this seems to be what happened

When HTMX process the following, 
```html
	<a href="#" hx-get="first.html" id="first">fist link</a>
	<a href="#" hx-get="second.html" hx-trigger="click from:#first">
        second link</a>
```
two `eventListerner` got registered, one for `hx-get` of `first link` itself, and one for `from:#first`, with different `triggerSpec`. When the the event is first handled, `eventData.handled` is set to `true`, and when the event listener is called again with a different `triggerSpec`, it is ignored due to 


https://github.com/bigskysoftware/htmx/blob/4cc29f9720dbedd0180902bfb8fa15abf0547cf5/src/htmx.js#L966

**Comment by BoPeng** [comment]
Just to confirm this is indeed the problem, the following patch fixes the problem although this does not look like a proper fix.

```diff
diff --git a/src/htmx.js b/src/htmx.js
index a261cda..21d7fbf 100644
--- a/src/htmx.js
+++ b/src/htmx.js
@@ -963,8 +963,8 @@ return (function () {
                 }
                 var eventData = getInternalData(evt);
                 var elementData = getInternalData(elt);
-                if (!eventData.handled) {
-                    eventData.handled = true;
+                if (eventData.handled !== triggerSpec) {
+                    eventData.handled = triggerSpec;
                     if (triggerSpec.once) {
                         if (elementData.triggeredOnce) {
                             return;
```

**Comment by 1cg** [comment]
I believe I have a fix for this in dev: https://github.com/bigskysoftware/htmx/commit/835436a99b12e980b484f820c7781f7b88d87d75

**Comment by BoPeng** [comment]
I can confirm that the patch works.

**Comment by 1cg** [comment]
fixed in 1.3.0
